Java/jni/native_view.cc

```diff
@@ -149,24 +149,26 @@ JNIEXPORT void JNICALL Java_com_couchbase_cbforest_View_emit(JNIEnv *env, jobjec
     jlong *keys   = env->GetLongArrayElements(jkeys, NULL);
     C4Key *c4keys[count];
     C4Slice c4values[count];
-    std::vector<jbyteArraySlice> valueBufs;
+    std::vector<jbyteArraySlice*> valueBufs;
     for(int i = 0; i < count; i++) {
         c4keys[i] = (C4Key*)keys[i];
         jbyteArray jvalue = (jbyteArray) env->GetObjectArrayElement(jvalues, i);
         if (jvalue) {
-            valueBufs.push_back(jbyteArraySlice(env, jvalue));
```

Comment:
Hm. The problem here is that the jbyteArraySlice is constructed and then copied into the vector, but then the original is destructed, which frees the JNI reference immediately.

I found a better way to do this without allocating memory — use `vector::emplace` which adds an element directly to the vector. Just replace line 157 with:

```c++
            valueBufs.emplace(valueBufs.end(), env, jvalue));
```
Then you don't need the rest of the changes to this method.